exonum/src/helpers/fabric/details.rs

```diff
@@ -459,8 +545,24 @@ impl Command for GenerateNodeConfig {
         let services_public_configs = new_context.get(keys::SERVICES_PUBLIC_CONFIGS).unwrap();
         let services_secret_configs = new_context.get(keys::SERVICES_SECRET_CONFIGS);
 
-        let (consensus_public_key, consensus_secret_key) = crypto::gen_keypair();
-        let (service_public_key, service_secret_key) = crypto::gen_keypair();
+        let consensus_passphrase = Self::get_passphrase(&new_context, EXONUM_CONSENSUS_PASS);
+        let service_passphrase = Self::get_passphrase(&new_context, EXONUM_SERVICE_PASS);
+        let consensus_public_key =
+            Self::create_secret_key_file(&consensus_secret_key_path, &consensus_passphrase);
+        let service_public_key =
+            Self::create_secret_key_file(&service_secret_key_path, &service_passphrase);
+
+        let pub_config_folder = Path::new(&pub_config_path).parent().unwrap();
+        let consensus_secret_key = if consensus_secret_key_path.is_absolute() {
+            consensus_secret_key_path
+        } else {
+            path_relative_from(&consensus_secret_key_path, &pub_config_folder).unwrap()
+        };
+        let service_secret_key = if service_secret_key_path.is_absolute() {
```

Comment:
It feels that it would be better to have this `is_absolute` branch as a part of `path_relative_from`.

I also believe that it will be more nice to read if the arguments switch places:

```rust
path_relative_from(&pub_config_folder, service_secret_key_path)
```

Reply:
I didn't want to edit the function which once was in the `stdlib`. I'd suggest to leave it as-is. It was somehow tested and reliable, I fell like it's better to keep that expected original behavior.

Reply:
If it were still around in stdlib as a deprecated function then I'd suggest using it instead. However, it was removed then I don't really buy the sentimental argument of keeping all the old quirks in naming, behavior, and argument order when we can do better now. It's not like new developers looking at this code will remember the old behavior. Plus, we have our own tests to verify that this function behaves as we expect it to.

Though, argument naming and order is a subjective matter so I'm most certainly not blocking this PR if you believe that we should keep the original naming and argument order.

Reply:
Yeah, I haven't brought up the main argument and I'm talking not against renaming the function right now.. I don't want to put the branch there. It really has nothing to do with the purpose of that function. The logic is to keep absolute paths as-is, because that's what user, I think, expects to see in the config if they pass an absolute path to the command line.

Reply:
I thing that current arguments named better since function is called "path relative from" - first argument is "path" second argument "from".
Regarding moving is_absolute branch into some function I would consider premature since there are only two use cases, also this thing with paths may change(see comments later)